### ISSUE TITLE
Allow options to have more aliases

### ISSUE THREAD

**Issue by felixSchl** [felixSchl/docopt.js#64]
Instead of

```
type Option = { name :: Maybe String, flag :: Maybe Char | ... }
```

it could be

```
data Alias = Long String | Short Char
type Option = { aliases :: NonEmpty Alias | ... }
```

This would allow to do things like `-h, -?, --help`

**Comment by felixSchl** [comment]
This work is almost complete and it really showed the strength of the type system. After swapping over it revealed many edge cases that had either no been dealt with at all or had been dealt with in a awkward way, since two `Maybe`s do not guarantee that one them will be a `Just`.

**Comment by felixSchl** [comment]
Landed via ede0daacd814a9989baec254203408d411aae544
